botocove/cove_decorator.py

```diff
@@ -64,26 +65,35 @@ def wrapper(*args: Any, **kwargs: Any) -> CoveOutput:
                 thread_workers=thread_workers,
             )
 
-            output = runner.run_cove_function()
+            function_output = runner.run_cove_function()
 
-            # Rewrite dataclasses into untyped dicts to retain current functionality
-            return CoveOutput(
+            cove_output = CoveOutput(
                 Results=[
                     {k: v for k, v in r.items() if v is not None}
-                    for r in output["Results"]
+                    for r in function_output["Results"]
                 ],
                 Exceptions=[
                     {k: v for k, v in e.items() if v is not None}
-                    for e in output["Exceptions"]
+                    for e in function_output["Exceptions"]
                     if e["AssumeRoleSuccess"] is True
                 ],
                 FailedAssumeRole=[
                     {k: v for k, v in f.items() if v is not None}
-                    for f in output["Exceptions"]
+                    for f in function_output["Exceptions"]
                     if f["AssumeRoleSuccess"] is False
                 ],
             )
 
+            if regions is None:
```

Comment:
I'm not following why this is required - I'm sure I'm missing something obvious but could you explain?

Reply:
Gotcha - makes sense. I don't think we need to maintain this behaviour in a new release: we can just always emit a Region key in the output and bump the sem ver appropriately. I'm not overly worried about adding to output, it's removing things that's more of a place to be wary of previous API contracts.